tap_fullstory/__init__.py

```diff
@@ -81,11 +81,10 @@ def on_giveup(details):
 
 @backoff.on_exception(backoff.expo,
                       requests.exceptions.RequestException,
-                      max_tries=5,
+                      max_tries=6,
                       giveup=giveup,
                       on_giveup=on_giveup,
                       factor=2)
-@utils.ratelimit(9, 1)
```

Comment:
This annotation in combination with the `@backoff.exception` annotation above was causing the backoff logic to behave weirdly, as if the number of current retries reset whenever we got up to 9 in this rate limiter. For FullStory's purposes, it's not strictly necessary to rate limit in this way since the exponential backoff waits appropriately when we get 429s. I don't know if there are other singer-related reasons to rate limit?

I considered changing this line to something like `@utils.ratelimit(2, 60)` to enforce the FullStory rate limit up front, but this would mean the tap couldn't make a burst of requests. For most daily ETL processes, the burst allowance should be sufficient to get all the export bundles for previous day without being rate limited, so it's important to allow bursting. I checked out the code in [utils.py](https://github.com/singer-io/singer-python/blob/master/singer/utils.py), but did not see any burst capabilities there.

Reply:
Actually FullStory does provide a well-formed `Retry-After` header, and I think this code should be rewritten to sleep for the suggested number of seconds. I'll do that when I have a chance to come back to this work properly.

Reply:
To be clear, I'll add the sleep after 429 as part of this PR, I'm just having a hard time getting a contiguous block of dev time with which to finish it.